controllers/spiaccesstoken_controller.go

```diff
@@ -134,7 +136,7 @@ func (r *SPIAccessTokenReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	ctx = infrastructure.InitKcpContext(ctx, req.ClusterName)
 
 	lg := log.FromContext(ctx)
-	defer logs.TimeTrack(lg, time.Now(), "Reconcile SPIAccessToken")
+	defer logs.TimeTrackWithLazyLogger(func() logr.Logger { return lg }, time.Now(), "Reconcile SPIAccessToken")
```

Comment:
We modify the lg instance during the course of the method by adding additional values, etc. Using the lazy approach, the logger as it exists at the exit from the method is used for the time track logging (i.e. it will contain all the key value pairs "accumulated" during the execution of the method), otherwise defer "captures" the `lg` as it was when defer was declared.